src/main/scala/mesosphere/marathon/state/PersistenceStore.scala

```diff
@@ -4,16 +4,17 @@ import scala.concurrent.Future
 
 trait PersistenceStore[T] {
 
+  type Read = () => T //returns deserialized T value.
```

Comment:
I think I would understand the intention better if this was called `Deserialize`